vcd/resource_vcd_nsxt_distributed_firewall_rule.go

```diff
@@ -0,0 +1,395 @@
+				Optional:    true,
+				Default:     false,
+				Description: "Reverses firewall matching for to match all except Destinations Groups specified in 'destination_ids' (VCD 10.3.2+)",
+			},
+		},
+	}
+}
+
+func resourceVcdNsxtDistributedFirewallRuleCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentVdcGroup(d)
+	defer vcdClient.unlockParentVdcGroup(d)
+
+	org, err := vcdClient.GetOrgFromResource(d)
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule create] error retriving Org: %s", err)
+	}
+
+	vdcGroup, err := org.GetVdcGroupById(d.Get("vdc_group_id").(string))
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule create] error retrieving VDC Group: %s", err)
+	}
+
+	firewallRuleType, err := getDistributedFirewallRuleType(vcdClient, d, "create")
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule create] error getting Distributed Firewall Rule type: %s", err)
+	}
+	_, singleRule, err := vdcGroup.CreateDistributedFirewallRule(d.Get("above_rule_id").(string), firewallRuleType)
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule create] error setting Distributed Firewall Rule: %s", err)
+	}
+
+	d.SetId(singleRule.Rule.ID)
+
+	return resourceVcdNsxtDistributedFirewallRuleRead(ctx, d, meta)
+}
+
+func resourceVcdNsxtDistributedFirewallRuleUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentVdcGroup(d)
+	defer vcdClient.unlockParentVdcGroup(d)
+
+	org, err := vcdClient.GetOrgFromResource(d)
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule update] error retriving Org: %s", err)
+	}
+
+	vdcGroup, err := org.GetVdcGroupById(d.Get("vdc_group_id").(string))
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule update] error retrieving VDC Group: %s", err)
+	}
+
+	rule, err := vdcGroup.GetDistributedFirewallRuleById(d.Id())
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule update] error retrieving Firewall Rule By ID: %s", err)
+	}
+
+	firewallRuleType, err := getDistributedFirewallRuleType(vcdClient, d, "create")
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule update] error getting Distributed Firewall Rule type: %s", err)
+	}
+
+	firewallRuleType.ID = rule.Rule.ID
+	_, err = rule.Update(firewallRuleType)
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule update] errorupdating Distributed Firewall Rule: %s", err)
+	}
+
+	return resourceVcdNsxtDistributedFirewallRuleRead(ctx, d, meta)
+}
+
+func resourceVcdNsxtDistributedFirewallRuleRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	org, err := vcdClient.GetOrgFromResource(d)
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule read] error retriving Org: %s", err)
```

Comment:
```suggestion
		return diag.Errorf("[Distributed Firewall Rule read] error retrieving Org: %s", err)
```

Reply:
Actually, this came from other place (old DFW resource). I did a search and replace for all such occurrences.